### PR TITLE
Framework dependent tests.

### PR DESCRIPTION
**Pull request by jaheba**
With these changes, `pytest` should suceed without having `mxnet` installed.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

File: test/core/test_component.py [hunk 1]
```diff
@@ -153,35 +153,6 @@ def compare_vals(x, y, z):
     assert baz01 == baz02
 
 
-def test_dynamic_loading():
```
Comment (mbohlkeschneider) [inline test/core/test_component.py:156]: Why is this removed? This is not moved anywhere, right? Anything else is moved to a specific `mx` test folder. Why is this not?

Comment (jaheba) [reply test/core/test_component.py:156]: Good spot! I've removed it since this also only seems to test `load_code` which is also tested elsewhere. I can add it to the mxnet tests though.

Comment (mbohlkeschneider) [reply test/core/test_component.py:156]: Not sure what the purpose is but I would keep it, if possible.

**Review by mbohlkeschneider — APPROVED** [review]
